extensions/fairygui/src/fairygui/display/BlendMode.cpp

```diff
@@ -0,0 +1,75 @@
+#include "BlendMode.h"
+#include "renderer/backend/Enums.h"
+
+NS_FGUI_BEGIN
+using namespace ax;
+
+namespace
+{
+std::vector<BlendFunc> blendModes {
+    {backend::BlendFactor::SRC_ALPHA, backend::BlendFactor::ONE_MINUS_SRC_ALPHA},  // normal
+    {backend::BlendFactor::ONE, backend::BlendFactor::ONE},                        // none
+    {backend::BlendFactor::SRC_ALPHA, backend::BlendFactor::ONE},                  // add
+    {backend::BlendFactor::DST_COLOR, backend::BlendFactor::ONE_MINUS_SRC_ALPHA},  // mul
+    {backend::BlendFactor::ONE, backend::BlendFactor::ONE_MINUS_SRC_COLOR},        // screen
+    {backend::BlendFactor::ZERO, backend::BlendFactor::ONE_MINUS_SRC_ALPHA},       // erase
+    {backend::BlendFactor::ZERO, backend::BlendFactor::SRC_ALPHA},                 // mask
+    {backend::BlendFactor::ONE_MINUS_DST_ALPHA, backend::BlendFactor::DST_ALPHA},  // below
+    {backend::BlendFactor::ONE, backend::BlendFactor::ZERO},                       // off
+    {backend::BlendFactor::SRC_ALPHA, backend::BlendFactor::ONE_MINUS_SRC_ALPHA},  // custom1
+    {backend::BlendFactor::SRC_ALPHA, backend::BlendFactor::ONE_MINUS_SRC_ALPHA},  // custom2
+    {backend::BlendFactor::SRC_ALPHA, backend::BlendFactor::ONE_MINUS_SRC_ALPHA},  // custom2
+};
+}
+
+void BlendModeUtils::apply(ax::Node* node, BlendMode blendMode)
+{
+    if (!node)
+        return;
+
+    auto blendIndex = static_cast<size_t>(blendMode);
+    if (blendIndex >= blendModes.size())
+    {
+        AXLOGE("[FairyGUI] Invalid BlendMode: {}", blendIndex);
+        return;
+    }
+
+    auto& blendFunc = blendModes[blendIndex];
+    apply(node, blendFunc);
+}
+
+void BlendModeUtils::apply(ax::Node* node, const ax::BlendFunc& blendFunc)
+{
+    auto blendProtocol = dynamic_cast<BlendProtocol*>(node);
+    if (blendProtocol)
+    {
+        auto& currentBlendFunc = blendProtocol->getBlendFunc();
+        if (currentBlendFunc.src != blendFunc.src || currentBlendFunc.dst != blendFunc.dst)
+        {
+            blendProtocol->setBlendFunc(blendFunc);
+        }
+    }
+
+    auto&& children = node->getChildren();
+    for (auto&& child : children)
+    {
+        apply(child, blendFunc);
+    }
+}
+
+void BlendModeUtils::override(BlendMode blendMode,
```

Comment:
should avoid use c++ keyword as symbol, maybe rename to `update`

Reply:
Yes, I noticed that, since I was just following their current implementation from [here](https://github.com/fairygui/FairyGUI-cocoscreator/blob/ccc3.0/source/src/display/BlendMode.ts).

I'll change it now to avoid the confusion.

Reply:
> Actually, how about `overrideBlendMode`, since it states the purpose of the method?

it's ok